target/linux/realtek/files-5.10/drivers/clk/realtek/clk-rtl83xx.h

```diff
@@ -0,0 +1,87 @@
+// SPDX-License-Identifier: GPL-2.0-only
+/*
+ * Realtek RTL83XX clock headers
+ * Copyright (C) 2022 Markus Stockhausen
+ */
+
+/*
+ * Several stuff
+ */
+
+#define RTL_SRAM_MARKER				(0x5EAF00D5)
+#define RTL_SRAM_BASE				(0x9F000000)
+
+/*
+ * Switch registers
+ */
+
+#define RTL_SW_CORE_BASE			(0xBB000000)
+
+#define RTL838X_PLL_GLB_CTRL			(0x0FC0)
+#define RTL838X_PLL_CPU_CTRL0			(0x0FC4)
+#define RTL838X_PLL_CPU_CTRL1			(0x0FC8)
+#define RTL838X_PLL_LXB_CTRL0			(0x0FD0)
+#define RTL838X_PLL_LXB_CTRL1			(0x0FD4)
+#define RTL838X_PLL_MEM_CTRL0			(0x0FDC)
+#define RTL838X_PLL_MEM_CTRL1			(0x0FE0)
+
+#define RTL839X_PLL_GLB_CTRL			(0x0024)
+#define RTL839X_PLL_CPU_CTRL0			(0x0028)
+#define RTL839X_PLL_CPU_CTRL1			(0x002C)
+#define RTL839X_PLL_LXB_CTRL0			(0x0038)
+#define RTL839X_PLL_LXB_CTRL1			(0x003C)
+#define RTL839X_PLL_MEM_CTRL0			(0x0048)
+#define RTL839X_PLL_MEM_CTRL1			(0x004C)
+
+#define RTL_PLL_CTRL0_CMU_SEL_PREDIV(v)		(((v) >> 0) & 0x3)
```

Comment:
any reason you are not adding defines for the 'offset' and masks?

```suggestion
#define RTL83XX_PLL_CTRL0_CMU_SEL_PREDIV_SHIFT  0
#define RTL83XX_PLL_CTRL0_CMU_SEL_PREDIV_MASK  GENMASK(1, 0)
#define RTL83XX_PLL_CTRL0_CMU_SEL_PREDIV(v) \
             (((v) & RTL83XX_PLL_CTRL0_CMU_SEL_PREDIV_MASK) >> RTL83XX_PLL_CTRL0_CMU_SEL_PREDIV_SHIFT)
```
yes it's a bit more typing, but it reads a lot better.
Also, swap your shift/mask direction, with GENMASK(endbit, startbit) having the advantage that it will match whatever the datasheet says baout which bits are being used. e.g. for the next one:

Reply:
As we include this header into the assembler too we should keep it simple.

Reply:
I don't doubt openwrt will be fine with that; but upstream probably prefes more clean more readable code in the end?

If you insist however, we can get it in first; then rework it and fix it later ;)